processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -251,6 +244,59 @@ public static boolean convertSegment(File toConvert, File converted, IndexSpec i
     public MMappedIndex mapDir(File inDir) throws IOException;
   }
 
+
+  public static boolean equalRowDims(Rowboat rb1, IndexableAdapter adapter1, Rowboat rb2, IndexableAdapter adapter2)
+  {
+    final int[][] dims1 = rb1.getDims();
+    final int[][] dims2 = rb2.getDims();
+    if(dims1.length != dims2.length){
+      return false;
+    }
+    final Indexed<String> dim1Names = adapter1.getDimensionNames();
```

Comment:
right now this also means we compare dimension names for every single row, is that necessary?